core/txpool/legacypool/celo_legacypool.go

```diff
@@ -0,0 +1,24 @@
+package legacypool
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// filter Filters transactions from the given list, according to remaining balance (per currency, minus l1Cost)
+// and gasLimit. Returns drops and invalid txs.
+func (pool *LegacyPool) filter(list *list, addr common.Address, l1Cost *big.Int, gasLimit uint64) (types.Transactions, types.Transactions) {
```

Comment:
You could also make this a member of `list`, as you're using many functions from it anyways.

Reply:
I'll rewrite it like that and see how it looks

Reply:
I didn't like how it looked by making it a member of the list, since I needed to pass some extra parameters from the pool.